From: https://github.com/MariaDB/server/commit/fafe60e7e2da72488ff3741367a962b45a23b8ee.patch
Author: Marko Mäkelä <marko.makela@mariadb.com>
Subject: [PATCH] MDEV-27134: Sporadic failure of DROP DATABASE test

Let us create and drop a separate database for getting rid of the
default database in the MDEV-22781 test.
---
 mysql-test/main/cte_nonrecursive.result | 5 +++--
 mysql-test/main/cte_nonrecursive.test   | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/mysql-test/main/cte_nonrecursive.result b/mysql-test/main/cte_nonrecursive.result
index b28c2d4e2065e..0d8bd151708a1 100644
--- a/mysql-test/main/cte_nonrecursive.result
+++ b/mysql-test/main/cte_nonrecursive.result
@@ -1693,7 +1693,9 @@ disconnect con1;
 #
 # MDEV-22781: create view with CTE without default database
 #
-drop database test;
+create database db;
+use db;
+drop database db;
 create database db1;
 create table db1.t1 (a int);
 insert into db1.t1 values (3),(7),(1);
@@ -1723,7 +1725,6 @@ a
 drop view db1.v1;
 drop table db1.t1;
 drop database db1;
-create database test;
 use test;
 #
 # MDEV-24597: CTE with union used multiple times in query
diff --git a/mysql-test/main/cte_nonrecursive.test b/mysql-test/main/cte_nonrecursive.test
index 41a0146e4d0db..0a18ff3a33369 100644
--- a/mysql-test/main/cte_nonrecursive.test
+++ b/mysql-test/main/cte_nonrecursive.test
@@ -1205,7 +1205,9 @@ DROP TABLE test.t;
 --echo # MDEV-22781: create view with CTE without default database
 --echo #

-drop database test;
+create database db;
+use db;
+drop database db;
 create database db1;
 create table db1.t1 (a int);
 insert into db1.t1 values (3),(7),(1);
@@ -1227,7 +1229,6 @@ drop view db1.v1;
 drop table db1.t1;
 drop database db1;

-create database test;
 use test;

 --echo #
